File: src/main/java/com/gymtracker/gym/templateExecrcises/repository/TemplateExerciseRepository.java
package com.gymtracker.gym.templateExecrcises.repository;

import com.gymtracker.gym.templateExecrcises.model.TemplateExercise;
import org.jspecify.annotations.NullMarked;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.Optional;


@NullMarked
@Repository
public interface TemplateExerciseRepository extends JpaRepository<TemplateExercise, Long> {

    Optional<TemplateExercise> findByIdAndWorkoutTemplate_UserId(Long id, Long userId);

    boolean existsByIdAndWorkoutTemplate_UserId(Long id, Long userId);

    void deleteByIdAndWorkoutTemplate_UserId(Long id, Long userId);


}

File: src/main/kotlin/com/gymtracker/gym/templateExercises/repository/TemplateExerciseRepository.kt
package com.gymtracker.gym.templateExercises.repository

import com.gymtracker.gym.templateExercises.model.TemplateExercise
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository

@Repository
interface TemplateExerciseRepository : JpaRepository<TemplateExercise, Long> {

    fun findByIdAndWorkoutTemplate_UserId(id: Long, userId: Long): TemplateExercise?

    fun existsByIdAndWorkoutTemplate_UserId(id: Long, userId: Long): Boolean

    fun deleteByIdAndWorkoutTemplate_UserId(id: Long, userId: Long)
}